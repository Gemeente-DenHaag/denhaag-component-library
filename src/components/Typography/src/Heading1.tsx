import React from 'react';
import BaseDataDisplayProps from '@gemeente-denhaag/basedatadisplayprops';
import './heading.css';
import clsx from 'clsx';

export type Heading1Props = BaseDataDisplayProps;

export const Heading1: React.FC<Heading1Props> = (props: Heading1Props) => {
  const rootClassNames = clsx('utrecht-heading-1 utrecht-heading-1--distanced', props.className);
  return <h1 className={rootClassNames}>{props.children}</h1>;
};
