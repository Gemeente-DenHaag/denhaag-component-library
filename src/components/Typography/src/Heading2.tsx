import React from 'react';
import BaseDataDisplayProps from '@gemeente-denhaag/basedatadisplayprops';
import './heading.css';
import clsx from 'clsx';

export type Heading2Props = Omit<BaseDataDisplayProps, 'classes'>;

export const Heading2: React.FC<Heading2Props> = (props: Heading2Props) => {
  const rootClassNames = clsx('utrecht-heading-2', 'utrecht-heading-2--distanced', props.className);
  return <h2 className={rootClassNames}>{props.children}</h2>;
};
