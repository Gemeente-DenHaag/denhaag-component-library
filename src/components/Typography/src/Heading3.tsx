import React from 'react';
import BaseDataDisplayProps from '@gemeente-denhaag/basedatadisplayprops';
import './heading.css';

export type Heading3Props = BaseDataDisplayProps;

export const Heading3: React.FC<Heading3Props> = (props: Heading3Props) => {
  return <h3 className="utrecht-heading-3 utrecht-heading-3--distanced">{props.children}</h3>;
};
