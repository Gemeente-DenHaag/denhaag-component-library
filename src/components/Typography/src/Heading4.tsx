import React from 'react';
import BaseDataDisplayProps from '@gemeente-denhaag/basedatadisplayprops';
import './heading.css';
import clsx from 'clsx';

export type Heading4Props = Omit<BaseDataDisplayProps, 'classes'>;

export const Heading4: React.FC<Heading4Props> = (props: Heading4Props) => {
  const rootClassNames = clsx('utrecht-heading-4 utrecht-heading-4--distanced', props.className);
  return <h4 className={rootClassNames}>{props.children}</h4>;
};
