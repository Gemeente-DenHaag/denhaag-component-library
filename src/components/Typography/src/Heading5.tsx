import React from 'react';
import BaseDataDisplayProps from '@gemeente-denhaag/basedatadisplayprops';
import './heading.css';

export type Heading5Props = BaseDataDisplayProps;

export const Heading5: React.FC<Heading5Props> = (props: Heading5Props) => {
  return <h5 className="utrecht-heading-5 utrecht-heading-5--distanced">{props.children}</h5>;
};
