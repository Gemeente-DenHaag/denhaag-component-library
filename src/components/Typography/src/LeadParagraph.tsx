import React from 'react';
import BaseDataDisplayProps from '@gemeente-denhaag/basedatadisplayprops';
import './paragraph.css';

export type LeadParagraphProps = BaseDataDisplayProps;

export const LeadParagraph: React.FC<LeadParagraphProps> = (props: LeadParagraphProps) => {
  return <p className="utrecht-paragraph utrecht-paragraph--lead utrecht-paragraph--distanced">{props.children}</p>;
};
