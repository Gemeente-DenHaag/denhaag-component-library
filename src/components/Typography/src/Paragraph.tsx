import React from 'react';
import BaseDataDisplayProps from '@gemeente-denhaag/basedatadisplayprops';
import './paragraph.css';
import { UtrechtParagraph } from '@utrecht/web-component-library-react';

export type ParagraphProps = Omit<BaseDataDisplayProps, 'classes'>;

export const Paragraph: React.FC<ParagraphProps> = (props: ParagraphProps) => {
  //return <p className={rootClassNames}> {props.children}</p>;
  return <UtrechtParagraph>{props.children}</UtrechtParagraph>;
};
