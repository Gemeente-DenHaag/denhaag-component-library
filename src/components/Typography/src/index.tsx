import React from "react";
import MaterialTypography from "@material-ui/core/Typography";
import BaseDataDisplayProps from "@gemeente-denhaag/basedatadisplayprops/src";

export interface TypographyProps extends BaseDataDisplayProps {
  /**
   * Set the text-align on the component.
   */
  align?: "inherit" | "left" | "center" | "right" | "justify";

  /**
   * The color of the component.
   * It supports those theme colors that make sense for this component.
   */
  color?: "initial" | "inherit" | "primary" | "secondary" | "textPrimary" | "textSecondary" | "error";

  /**
   * Controls the display type
   */
  display?: "initial" | "block" | "inline";

  /**
   * If true, the text will have a bottom margin.
   */
  gutterBottom?: boolean;

  /**
   * If true, the text will not wrap,
   * but instead will truncate with a text overflow ellipsis.
   */
  noWrap?: boolean;

  /**
   * If true, the text will have a bottom margin.
   */
  paragraph?: boolean;

  /**
   * Applies the theme typography styles.
   */
  variant?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "subtitle1"
    | "subtitle2"
    | "body1"
    | "body2"
    | "caption"
    | "button"
    | "overline"
    | "srOnly"
    | "inherit";

  /**
   * The component maps the variant prop to a range of different HTML element types.
   */
  variantMapping?: Record<string, unknown>;
}

/**
 * Allows developers to present and design content clearly and efficiently.
 * @param props The properties of a Typography component.
 * @constructor Constructs an instance of Typography.
 */
export const Typography: React.FC<TypographyProps> = (props: TypographyProps) => {
  return <MaterialTypography {...props}>{props.children}</MaterialTypography>;
};

export default Typography;
