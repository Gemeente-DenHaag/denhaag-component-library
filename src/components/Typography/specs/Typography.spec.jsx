import * as React from 'react';
import { Heading1, Heading2, Heading3, Heading4, Heading5, Paragraph, LeadParagraph } from '../src/index';

describe(`Typography tests`, () => {
  it('a11y and snapshots', () => {
    cy.snapshots(<Heading1>This is a bit of text inside a Heading1 component.</Heading1>, {});
    cy.snapshots(<Heading2>This is a bit of text inside a Heading2 component.</Heading2>, {});
    cy.snapshots(<Heading3>This is a bit of text inside a Heading3 component.</Heading3>, {});
    cy.snapshots(<Heading4>This is a bit of text inside a Heading4 component.</Heading4>, {});
    cy.snapshots(<Heading5>This is a bit of text inside a Heading5 component.</Heading5>, {});
    cy.snapshots(<Paragraph>This is a bit of text inside a Paragraph component.</Paragraph>, {});
    cy.snapshots(<LeadParagraph>This is a bit of text inside a LeadParagraph component.</LeadParagraph>, {});
  });
});
