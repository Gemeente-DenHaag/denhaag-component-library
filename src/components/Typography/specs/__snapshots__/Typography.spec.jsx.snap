exports[`Typography tests > a11y and snapshots #0`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <h1 class="utrecht-heading-1 utrecht-heading-1--distanced">
      This is a bit of text inside a Heading1 component.
    </h1>
  </div>
</div>
`;

exports[`Typography tests > a11y and snapshots #1`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <h2 class="utrecht-heading-2 utrecht-heading-2--distanced">
      This is a bit of text inside a Heading2 component.
    </h2>
  </div>
</div>
`;

exports[`Typography tests > a11y and snapshots #2`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <h3 class="utrecht-heading-3 utrecht-heading-3--distanced">
      This is a bit of text inside a Heading3 component.
    </h3>
  </div>
</div>
`;

exports[`Typography tests > a11y and snapshots #3`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <h4 class="utrecht-heading-4 utrecht-heading-4--distanced">
      This is a bit of text inside a Heading4 component.
    </h4>
  </div>
</div>
`;

exports[`Typography tests > a11y and snapshots #4`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <h5 class="utrecht-heading-5 utrecht-heading-5--distanced">
      This is a bit of text inside a Heading5 component.
    </h5>
  </div>
</div>
`;

exports[`Typography tests > a11y and snapshots #5`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <p class="utrecht-paragraph utrecht-paragraph--distanced">
      This is a bit of text inside a Paragraph component.
    </p>
  </div>
</div>
`;

exports[`Typography tests > a11y and snapshots #6`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <p
      class="utrecht-paragraph utrecht-paragraph--lead utrecht-paragraph--distanced"
    >
      This is a bit of text inside a LeadParagraph component.
    </p>
  </div>
</div>
`;
